idea-plugin/src/main/kotlin/io/github/composegears/valkyrie/processing/writter/FileWriter.kt

```diff
@@ -13,8 +14,13 @@ object FileWriter {
         content: String,
         outDirectory: String,
         fileName: String,
+        deleteIfExists: Boolean = true,
     ) {
-        val outPath = Path("$outDirectory/$fileName.kt")
+        val outPath = Path(outDirectory, "$fileName.kt")
+
+        if (deleteIfExists && outPath.exists()) {
+            outPath.deleteIfExists()
+        }
```

Comment:
And we can add a param `createParent: Boolean = true,` for checking `outPath.createParentDirectories()`, this is helpful in loops.

Reply:
Or it could be addressed in a follow-up.

Reply:
will add